solution_template/jenkins/nested/storageAccount-existing.json

```diff
@@ -0,0 +1,39 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "outputs": {
+    "primaryEndpointBlob": {
```

Comment:
For the public IP output and storage account output - why don't you just reference the value itself in the main template?

Rather than:
`reference(variables('storageDeploymentName')).outputs.primaryEndpointBlob.value`
Just do:
`reference(resourceId('Microsoft.Storage/storageAccounts',parameters('storageAccountName')),'2016-01-01').primaryEndpoints.blob`